boot/bootutil/src/image_validate.c

```diff
@@ -449,44 +464,55 @@ bootutil_img_validate(struct enc_key_data *enc_state, int image_index,
              */
             if (len != sizeof(img_security_cnt)) {
                 /* Security counter is not valid. */
-                return -1;
+                rc = -1;
+                goto out;
             }
 
             rc = LOAD_IMAGE_DATA(hdr, fap, off, &img_security_cnt, len);
             if (rc) {
-                return rc;
+                goto out;
             }
 
-            rc = boot_nv_security_counter_get(image_index, &security_cnt);
-            if (rc) {
-                return rc;
+            FIH_CALL(boot_nv_security_counter_get, fih_rc, image_index,
+                                                           &security_cnt);
+            if (fih_not_eq(fih_rc, FIH_SUCCESS)) {
+                goto out;
             }
 
             /* Compare the new image's security counter value against the
              * stored security counter value.
              */
-            if (img_security_cnt < security_cnt) {
-                /* The image's security counter is not accepted. */
-                return -1;
+            fih_rc = fih_int_encode_zero_equality(img_security_cnt <
```

Comment:
```suggestion
  if (img_security_cnt < fih_int_decode(security_cnt)) {
    fih_rc = FIH_SUCCESS;
  } else {
    fih_rc = FIH_FAILURE;
  }
```
just think that usage _encode_ funct here is overcomplicated

Reply:
Yeah, tend to agree.  I think both version has the same level of resistance against FI. With the usage of `fih_int_encode_zero_equality` the code is a bit more concise and if  `fih_int_encode_zero_equality` would have adding some random delay (calling `fih_delay()`) then it would be more robust. But I can accept your version as well. Let me know which one do you prefer.

Reply:
This was not a strong suggestion from my side.